impl/src/main/kotlin/com/android/declarative/internal/configurators/IncludedBuildPluginCache.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.declarative.internal.configurators
+
+import org.gradle.api.Project
+import java.io.File
+import java.util.Properties
+
+class IncludedBuildPluginInfo(
+    val classesDir: File,
+    val resourcesDir: File,
+)
+
+class IncludedBuildPluginCache(
+    private val project: Project,
+    private val includedBuildDir: File,
+) {
+    private val markerFileLookup = "**/META-INF/gradle-plugins/*.properties"
+
+    private val markerFiles: Set<File> =
+        project.fileTree(includedBuildDir) {
+            it.include(markerFileLookup)
+        }.files
+
+
+    fun getPluginClassPath(): Set<File> =
+        mutableSetOf<File>().also { allFolders ->
+            markerFiles.forEach { markerFile ->
+                getBinaryFolders(markerFile).let {
+                    allFolders.add(it.first)
+                    allFolders.add(it.second)
+                }
+            }
+        }
+
+
+    fun resolvePluginById(id:String): IncludedBuildPluginInfo? {
```

Comment:
Hm, cannot find resolvePluginById usage.

Reply:
there is none so far, I used it earlier but kept it even if unused as it might be needed later.